diff --git a/src/Plugin/Condition/Php.php b/src/Plugin/Condition/Php.php
index a803a49..dd95f79 100644
--- a/src/Plugin/Condition/Php.php
+++ b/src/Plugin/Condition/Php.php
@@ -65,7 +65,11 @@ class Php extends ConditionPluginBase {
    * {@inheritdoc}
    */
   public function evaluate() {
-    return php_eval($this->configuration['php']);
+    if (isset($this->configuration['php']) && !empty($this->configuration['php']) && trim($this->configuration['php']) !== '' ) {
+      return php_eval($this->configuration['php']);
+    }
+    else{
+      return true;
+    }
   }
-
 }
